tests/schemer/schema_test.py

```diff
@@ -351,3 +352,34 @@ def test_default_value_does_not_overwrite_existing(self):
         blog_post_schema.apply_defaults(self.document_1)
         self.assertEqual(35, self.document_1['likes'])
         self.assertEqual(datetime(1980, 5, 3), self.document_1['creation_date'])
+
+class TestSchemaLevelValidators(unittest.TestCase):
+    def test_schema_level_validator_single_item_failure(self):
+        doc = {'has_json': True}
+        with self.assertRaises(ValidationException):
+            validator_schema_1.validate(doc)
```

Comment:
These tests should also test that we're reporting the validation errors correctly. Check out the `assert_document_paths_invalid` function above and how that's used to quickly check we're getting the right errors raised.